OpenPGP provider and status fixes

Valodim opened the pull request:
The two commits in this PR fix #1150 and #1151, respectively

philipwhiuk commented:
Does it also fix #1153 and #1160 ?

Valodim commented:
It doesn't.

philipwhiuk commented:
Ok no worries.

cketti commented:
Still doesn't work for me without a crypto provider configured.

philipwhiuk commented:
With my PR as well it works for me.

cketti commented:
As far as I can tell `ComposeCryptoStatus.shouldUsePgpMessageBuilder()` returns `true` because the default 'crypto mode' of 'opportunistic' is used. Whether or not a crypto provider is configured doesn't factor into the decision.

Valodim commented:
Haha, I fixed it in my first commit, then broke it in the second. Derp.

Working on this right now.

Valodim commented:
Alrighty, that should be fixed. I also moved all openpgpservicconnection handling into the presenter, and added better handling for a lost service connection. I tested a couple of scenarios while installing and deinstalling openkeychain all around, and it seems more robust to me now.

I wonder what trivial usage scenario I missed this time ;)

philipwhiuk commented on k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java:
```diff
@@ -3022,7 +2995,8 @@ public void onMessageBuildCancel() {
-        Toast.makeText(MessageCompose.this, getString(R.string.send_aborted), Toast.LENGTH_LONG).show();
+        Toast.makeText(MessageCompose.this,
+                getString(R.string.send_failed_reason, me.getLocalizedMessage()), Toast.LENGTH_LONG).show();
         currentMessageBuilder = null;
```
I'm not convinced by this - it's not like the actual "send" (i.e. the I/O) actually happens.

cketti replied:
Please don't display exception messages to the user. We need better error reporting, but that's not what I had in mind :smile: 

Maybe "Send aborted due to an error"?

Valodim replied:
I did this because otherwise messages like "Signing key not found" would not get through to the user, giving them no clue whatsoever what was wrong other than "Send aborted"...

cketti replied:
I figured. But what we really want is an error code -> error message mapping inside K-9 Mail. That way we can provide appropriate error messages that can be localized by our translators.

Valodim replied:
Right. We already have a class for openpgp api errors, which also does error codes. We don't really use that in OpenKeychain yet but I have some ideas how to do that from our internal error logs. I also wanted to pass a pending intent result which shows the log we keep internally, which may be helpful for debugging.

Anyways, I reckon I'll wait with that until there is better error handling in k9 first.

philipwhiuk commented:
> Haha, I fixed it in my first commit, then broke it in the second. Derp.

It's annoying this is all front-end code. It would be nice get some UTs on it.
